src/common/components/game/GameArea.js

```diff
@@ -0,0 +1,282 @@
+  attack: object,
+
+  blueHand: arrayOf(object),
+  orangeHand: arrayOf(object),
+
+  bluePieces: object,
+  orangePieces: object,
+
+  blueEnergy: object,
+  orangeEnergy: object,
+
+  blueDeck: arrayOf(object),
+  orangeDeck: arrayOf(object),
+
+  blueDiscardPile: arrayOf(object),
+  orangeDiscardPile: arrayOf(object),
+
+  eventQueue: arrayOf(object),
+  sfxQueue: arrayOf(string),
+  tutorialStep: object,
+
+  history: object,
+  location: object,
+  match: object,
+
+  gameOver: bool,
+  isTutorial: bool,
+  isMyTurn: bool,
+  isSpectator: bool,
+  isAttackHappening: bool,
+
+  actionLog: arrayOf(object),
+  socket: object
+};
+
+export default class GameArea extends Component {
+  /* eslint-disable react/no-unused-prop-types */
+  static propTypes = {
+    ...gameProps,
+
+    chatOpen: bool,
+    message: string,
+
+    onPassTurn: func,
+    onForfeit: func,
+    onTutorialStep: func,
+    onSendChatMessage: func,
+    onActivateObject: func,
+    onClickGameArea: func,
+    onClickEndGame: func,
+    onNextTutorialStep: func,
+    onPrevTutorialStep: func,
+    onSelectTile: func
+  };
+  /* eslint-enable react/no-unused-prop-types */
+
+  state = {
+    areaHeight: 1250,
+    boardSize: 1000,
+    chatOpen: true
+  };
+
+  componentDidMount() {
+    this.updateDimensions();
+    window.addEventListener('resize', () => { this.updateDimensions(); });
+  }
+
+  handleToggleFullScreen = () => {
+    screenfull.toggle(this.gameArea);
+  };
+
+  handleToggleChat = () => {
+    this.setState(state => ({ chatOpen: !state.chatOpen }));
+  };
+
+  loadBackground = () => inBrowser() ? require('../img/black_bg_lodyas.png') : '';
+
+  updateDimensions = () => {
+    const maxBoardHeight = window.innerHeight - 64 - 150;
+    const maxBoardWidth = window.innerWidth - 256;
+
+    this.setState({
+      areaHeight: window.innerHeight - 64,
+      boardSize: Math.min(maxBoardWidth, maxBoardHeight)
+    });
+  };
+
+  render() {
+    const {
+      actionLog, attack, bluePieces, currentTurn, eventQueue, gameOver, isAttackHappening,
+      isMyTurn, isSpectator, isTutorial, message, orangePieces, player, playingCardType,
+      selectedTile, sfxQueue, socket, status, target, tutorialStep, usernames, winner,
+      onActivateObject, onClickEndGame, onClickGameArea, onForfeit, onNextTutorialStep,
+      onPassTurn, onPrevTutorialStep, onSelectTile, onSendChatMessage, onTutorialStep
+    } = this.props;
+    const { areaHeight, boardSize, chatOpen } = this.state;
+
+    if (message) {
+      return <FullscreenMessage message={message} height={areaHeight} background={this.loadBackground()} />;
+    }
+
+    return (
+      <div
+        id="gameArea"
+        className="gameArea"
+        ref={(gameArea) => { this.gameArea = gameArea; }}
+        style={
+          screenfull.isFullscreen ? {width: '100%', height: '100%'} : {}
+        }
+      >
+        <div>
+          <GameNotification text="It's your turn!" enabled={currentTurn === player} />
```

Comment:
[nit] This might be overkill, but should we maybe have a `isPlayersTurn` function on the container?